### PR TITLE
Remove void casts and use `[[maybe_unused]]` instead

### PR DESCRIPTION
**Pull request by OresteMarquis**
deal.ii workshop tutorials on PR, tria.cc file (first 5-10 appearance changed to [[maybe_unused]])

This is for the issue #17247 .

### CODE REVIEW

**Comment by bangerth** [comment]
Actually, for #17247.

File: source/grid/tria.cc [hunk 1]
```diff
@@ -1066,7 +1066,6 @@ namespace internal
       {
         (void)global_first_cell;
         (void)global_num_cells;
```
Comment (blaisb) [inline source/grid/tria.cc:1068]: It would be great to address both of these variables while you are at it :)

Comment (OresteMarquis) [reply source/grid/tria.cc:1068]: I was suppose to do only 5-10, other people are doing the other ones. There are actually 23 occurence of (void)x in this file.

Comment (blaisb) [reply source/grid/tria.cc:1068]: Okidoke :) Sorry about that it's my bad.

**Review by blaisb — APPROVED** [review]

**Review by gassmoeller — APPROVED** [review]
With #17524 we can merge this now and fix the broken gcc afterwards.

**Pull request merged**